### ISSUE TITLE
[SYCL][HIP] Required aspect fp16 is not supported on the device

### ISSUE THREAD

**Issue by zjin-lcf** [intel/llvm#8330]
Running a SYCL program that contains "half" data types on a MI100 GPU shows the error:  Required aspect fp16 is not supported on the device.  

Please see a reproducer. Can you reproduce the error using a recent build of the SYCL compiler with HIP support ?  Thanks.

https://github.com/zjin-lcf/HeCBench/tree/master/f16sp-sycl

**Comment by npmiller** [comment]
Yes, that's expected, half support for HIP is currently disabled, we need to look into it, see related comment:

* https://github.com/intel/llvm/pull/4223#issuecomment-890005869

**Comment by zjin-lcf** [comment]
I am confused that the tahiti architecture includes GPUs of the vega20 architecture when browsing the CMakefileList.txt in the libclc. 

```
# Setup aliases
set( cedar_aliases palm sumo sumo2 redwood juniper )
set( cypress_aliases hemlock )
set( barts_aliases turks caicos )
set( cayman_aliases aruba )
set( tahiti_aliases pitcairn verde oland hainan bonaire kabini kaveri hawaii
        mullins tonga iceland carrizo fiji stoney polaris10 polaris11 )

# Support for gfx9 was added in LLVM 5.0 (r295554)
if( ${LLVM_VERSION} VERSION_GREATER "4.99.99" )
        set( tahiti_aliases ${tahiti_aliases} gfx900 gfx902 )
endif()

# Support for Vega12 and Vega20 was added in LLVM 7 (r331215)
if( ${LLVM_VERSION} VERSION_GREATER "6.99.99" )
        set( tahiti_aliases ${tahiti_aliases} gfx904 gfx906 )
endif()
```

**Comment by npmiller** [comment]
Oooh, yes, the general support for fp16 was merged but adding the aspect to the HIP plugin hasn't:

* https://github.com/oneapi-src/unified-runtime/pull/988

So it will say it won't support it but it actually does, if you remove the aspect check from the program it should work fine.

**Comment by zjin-lcf** [comment]
There's no aspect check in user programs.  Thanks.
